web/src/activity_ui.js

```diff
@@ -137,6 +137,15 @@ export function initialize(opts) {
 }
 
 export function update_presence_info(user_id, info, server_time) {
+    // There can be some case where the presence event
+    // was set for an inaccessible user if
+    // CAN_ACCESS_ALL_USERS_GROUP_LIMITS_PRESENCE is
+    // disabled. We just ignore that event and return.
+    const person = people.maybe_get_user_by_id(user_id, true);
+    if (person === undefined || person.is_inaccessible_user) {
+        return;
```

Comment:
I have not added checks here like `settings_data.user_can_access_all_other_users` or check for `CAN_ACCESS_ALL_USERS_GROUP_LIMITS_PRESENCE` assuming that this would only happen for an inaccessible user.

Reply:
Can you talk about what motivates a change here at all (i.e. what happens if we fall through?)